Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -578,6 +613,127 @@ setActivity (personId, _) isActive mode = do
       DMode.getDriverStatus mode isActive
   pure APISuccess.Success
 
+activateGoHomeFeature :: (CacheFlow m r, EsqDBFlow m r) => (Id SP.Person, Id DM.Merchant) -> Id DDHL.DriverHomeLocation -> m APISuccess.APISuccess
+activateGoHomeFeature (driverId, merchantId) driverHomeLocationId = do
+  _ <- QPerson.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
+  goHomeConfig <- CQGHC.findByMerchantId merchantId
+  unless (goHomeConfig.enableGoHome) $ throwError GoHomeFeaturePermanentlyDisabled
+  driverInfo <- QDriverInformation.findById driverId >>= fromMaybeM DriverInfoNotFound
+  unless driverInfo.enabled $ throwError DriverAccountDisabled
+  unless (not driverInfo.blocked) $ throwError DriverAccountBlocked
+  driverLocation <- QDrLoc.findById driverId >>= fromMaybeM LocationNotFound
+  let currPos = LatLong {lat = driverLocation.lat, lon = driverLocation.lon}
+  driverHomeLocation <- QDHL.findById driverHomeLocationId >>= fromMaybeM (DriverHomeLocationDoesNotExist driverHomeLocationId.getId)
+  let homePos = LatLong {lat = driverHomeLocation.lat, lon = driverHomeLocation.lon}
+  unless (distanceBetweenInMeters homePos currPos > fromIntegral goHomeConfig.destRadiusMeters) $ throwError DriverCloseToHomeLocation
+  dghInfo <- CQDGR.getDriverGoHomeRequestInfo driverId merchantId
+  whenM (liftM2 (||) (return $ isJust dghInfo.status) (isJust <$> QDGR.findActive driverId)) $ throwError DriverGoHomeRequestAlreadyActive
+  unless (dghInfo.cnt > 0) $ throwError DriverGoHomeRequestDailyUsageLimitReached
+  activateDriverGoHomeRequest merchantId driverId driverHomeLocation
+  --QDGR.create =<< buildDriverGoHomeRequest driverHomeLocation
+  pure APISuccess.Success
+
+-- where
+--   buildDriverGoHomeRequest driverHomeLocation = do
+--     id <- generateGUID
+--     now <- getCurrentTime
+--     return $
+--       DDGR.DriverGoHomeRequest
+--         { lat = driverHomeLocation.lat,
+--           lon = driverHomeLocation.lon,
+--           status = DDGR.ACTIVE,
+--           createdAt = now,
+--           updatedAt = now,
+--           ..
+--         }
+
+deactivateGoHomeFeature :: (CacheFlow m r, EsqDBFlow m r) => (Id SP.Person, Id DM.Merchant) -> m APISuccess.APISuccess
+deactivateGoHomeFeature (personId, merchantId) = do
+  _ <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
```

Comment:
this can be written like this
void $ QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)